internal/services/autopi/hardware_template_service.go

```diff
@@ -149,8 +150,26 @@ func (a *hardwareTemplateService) ApplyHardwareTemplate(ctx context.Context, req
 	if autoPi.Template > 0 {
 		err = a.ap.UnassociateDeviceTemplate(autoPi.ID, autoPi.Template)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to unassociate template")
+			return nil, errors.Wrap(err, fmt.Sprintf("failed to unassociate template %d on device %s", autoPi.Template, autoPi.ID))
 		}
+
+		var vehicleLoggers *[]services.AutoPiVehicleLogger
```

Comment:
remove unnecessary line to instantiate vehicleLoggers, just assign it inline on next line.

Reply:
done